src/rdb.c

```diff
@@ -1970,24 +1968,52 @@ robj *rdbLoadObject(int rdbtype, rio *rdb, sds key, int dbid, int *error) {
                     zsetConvert(o,OBJ_ENCODING_SKIPLIST);
                 break;
             case RDB_TYPE_HASH_ZIPLIST:
+                {
```

Comment:
i see in here, we didn't check `deep_integrity_validation`
and `hashZiplistConvertAndValidateIntegrity` now rename to `ziplistPairsConvertAndValidateIntegrity`
it always pass `deep = 1` to `ziplistValidateIntegrity`

Is this intentional or is it an overlook?
we should check `deep_integrity_validation` or need a `stat_dump_payload_sanitizations++`?

Reply:
Yes, because ziplist is converted to listpack, so we have to do a deep sanitization, otherwise it will be wrong in subsequent operations as well.

Reply:
it's a decision we previously took. if anyway we're gonna do O(N) operation on conversion, we might as well do O(2N), and avoid complicated code.